ui/src/importer/importerTaskDetails/ImportedDocuments.js

```diff
@@ -301,83 +153,91 @@ export class ImportedDocuments extends React.Component {
     });
   };
 
-  renderCancelButton = () => {
-    const { isLoading } = this.state;
-    const { taskId } = this.props;
-
-    return (
-      isLoading && (
-        <>
-          <CancelImportTask logId={taskId} />{' '}
-          <Icon loading name="circle notch" />
-          This may take a while. You may leave the page, the process will
-          continue in background. ImportedSearch
-        </>
-      )
-    );
-  };
-
   render() {
+    const { data, taskId, isLoading, error } = this.props;
     const {
-      data,
       statistics,
       selectedResult,
       filteredRecords,
       activePage,
     } = this.state;
+
+    const dataAvailable = !_isEmpty(data);
+    const filteredRecordsAvailable = !_isEmpty(filteredRecords);
+    const filteredRecordsNotNull = !_isNull(filteredRecords);
+    const errorWhileFetching = !_isEmpty(data) && data?.status === 'FAILED';
+    const entriesReady =
+      (data?.loaded_entries || data?.loaded_entries === 0) &&
+      data?.entries_count;
+
+    if (error) {
```

Comment:
the not found will be displayed on any error in this case, shouldn't be only for 404?

Reply:
That is true but i thought about some errors that could happen and i felt like it could serve multiple purposes, it just simplifies the code and logic.